articles/contributing/code.md

```diff
@@ -0,0 +1,100 @@
+For instance, the following function is used to ensure that the <xref:microsoft.quantum.canon.fst> and <xref:microsoft.quantum.canon.snd> functions both return the right outputs in a representative example.
+If the output of `Fst` or `Snd` is incorrect, the `fail` statement is used to cause the test to fail.
+
+```qsharp
+function PairTest () : Unit {
+    let pair = (12, PauliZ);
+
+    if (Fst(pair) != 12) {
+        let actual = Fst(pair);
+        fail $"Expected 12, actual {actual}.";
+    }
+
+    if (Snd(pair) != PauliZ) {
+        let actual = Snd(pair);
+        fail $"Expected PauliZ, actual {actual}.";
+    }
+}
+```
+
+More complicated conditions can be checked using the techniques in the [testing section](xref:microsoft.quantum.libraries.standard.testing) of the standard libraries guide.
+For instance, the following test checks that `H(q); X(q); H(q);` as called by <xref:microsoft.quantum.canon.applywith> does the same thing as `Z(q)`.
+
+```qsharp
+operation WithTest () : Unit {
+    let actual = ApplyWith(H, X, _);
+    let expected = Z;
+    AssertOperationsEqualReferenced(ApplyToEach(actual, _), ApplyToEachA(expected, _), 4);
+}
+```
+
+When adding new features, it's a good idea to also add new tests to make sure that your contribution does what it's supposed to.
+This helps the rest of the community to maintain and develop your feature, and in particular helps other developers know that they can rely on your feature.
+
+> [!NOTE]
+> This works the other way around, too!
+> If there's an existing feature that's missing some tests, helping us add test coverage would make a great contribution to the community.
+
+Locally, unit tests can be run using the Visual Studio Test Explorer or the `dotnet test` command, so that you can check your contribution before opening up a pull request.
+
+<!-- TODO:
+### Comments and Documentation ###
+
+### Citations and References ### -->
+
+## When We'll Reject a Pull Request ##
+
+Sometimes, we'll reject the pull request for a contribution.
+If this happens to you, it doesn't mean that it's bad, as there's a number of reasons why we might not be able to accept a particular contribution.
+Perhaps most commonly, a contribution to the quantum programming community is a really good one, but the Quantum Development Kit repositories aren't the right place to develop it.
+In such cases, we strongly encourage you to make your own repository --- part of the strength of the Quantum Development Kit is that it's easy to make and distribute your own libraries using GitHub and NuGet.org, the same way that we distribute the canon and chemistry libraries today.
+
+At other times, we may reject a good contribution simply because we aren't yet ready to maintain and develop it.
+It can be difficult to do everything, so we plan out what features we're best able to work on as a roadmap.
+This can be another case where releasing a feature as a third-party library can make a lot of sense.
```

Comment:
As a note for the future, it would be cool to point to Q# community organization here, but it is probably not ready for that yet.

Reply:
@tcNickolas - definitely!

Reply:
I think that makes a lot of sense, @tcNickolas. I'll look forward to adding the qsharp.community link in the future, then.